guides/hack/08-generic-types-and-functions/20-reified-generics.md

```diff
@@ -0,0 +1,187 @@
+# Reified Generics
```

Comment:
What are the interactions between not reified generics and reified ones? What happens when a lie is passed? F.e. when the type checker thinks something is a `Box of int`, but at runtime this turns out to be false. Are type checks preformed on arguments to functions with reified generic parameters?

```HACK
class Ref<reify T>{ public function __construct(public T $v) {} }
class Box<T>{public function __construct(public T $v) {} }
function get_tany(mixed $mixed){ return $mixed; }

//Would this be allowed? 
$b = new Box(1);
$r = new Ref<int>($b->v);

//Or this?
function lie(Box<int> $boxed_int): void{
  $_ = new Ref<int>($boxed_int->v);
}
lie(new Box(get_tany('not an int')));
```